packages/appeals-service-api/src/routes/v2/appeal-cases/_caseReference/appeal-final-comments/repo.js

```diff
@@ -0,0 +1,86 @@
+const { createPrismaClient } = require('#db-client');
+const { PrismaClientKnownRequestError } = require('@prisma/client/runtime/library');
+
+/**
+ * @typedef {import('@prisma/client').AppealFinalComment} AppealFinalComment
+ */
+class AppealFinalCommentRepository {
+	dbClient;
+
+	constructor() {
+		this.dbClient = createPrismaClient();
+	}
+
+	/**
+	 * Get lpa statement for a given case reference
+	 *
+	 * @param {string} caseReference
+	 * @returns {Promise<Array<AppealFinalComment>|null>}
+	 */
+	async getLPAFinalComments(caseReference) {
+		try {
+			return await this.dbClient.appealFinalComment.findMany({
+				where: {
+					caseReference,
+					lpaCode: { not: null }
+				},
+				include: {
+					FinalCommentDocuments: {
+						include: {
+							Document: true
```

Comment:
Can we reuse 'DocumentsArgsPublishedOnly' here

Reply:
The DocumentsArgsPublishedOnly isnt reusable here as Prisma wont allow a where clause on the indirect Document relation.

Reply:
Created a query which is essentially the same but allows going through the indirect query

Reply:
couldn't it be reused but just a level lower? Or is that what you tried?
```js
include: {
  FinalCommentDocuments: {
    include: {
	    Document: {
	      where: ...
	      select: ...
	    }
```

Reply:
I guess that wouldn't filter the join results, this is fine I guess, would just be nice to have the required props/where for display reused

Reply:
That was what i tried, from what i found Prisma doesnt like that query as its using a where clause on an indirect relation. What ive put in works on the indirect query